pkg/flow/adapter/transformation/adapter.go

```diff
@@ -246,5 +245,9 @@ func (t *adapter) applyTransformations(event cloudevents.Event) (*cloudevents.Ev
 		t.logger.Errorw("Event transformation errors", zap.Errors("errors", errs))
 	}
 
+	// remove event-related variables after the transformation is done.
+	// since the storage is shared, flush can be done for one pipeline.
+	t.ContextPipeline.Storage.Flush(eventUniqueID)
```

Comment:
There might be some errors being raised before reaching this, would probably be better to use defer after line 207. I think even if the `apply` methods return error, the storage might be left with dangling items.

```go
	eventUniqueID := fmt.Sprintf("%s-%s", event.ID(), event.Source())
	// remove event-related variables after the transformation is done.
	// since the storage is shared, flush can be done for one pipeline.
	defer t.ContextPipeline.Storage.Flush(eventUniqueID)

	// Run init step such as load Pipeline variables first
	eventContext, err := t.ContextPipeline.apply(eventUniqueID, localContextBytes, init)
```

Reply:
done in 7d1a158